DataDog/cloudformation-template
# Update AWS quickstart cloudformation template to pass in aws_partition to v1 AWS API Create V2

### What does this PR do?

Update AWS quickstart cloudformation template to pass in aws_partition to v1 AWS API Create

This is to fix issue for role based accounts in gov cloud as we previously defaulted using the commercial partition. This change will pass in the partition associated with the account so should fix account creation for gov AWS accounts

### Motivation

This is to fix issue for role based accounts in gov cloud as we previously defaulted using the commercial partition. This change will pass in the partition associated with the account so should fix account creation for gov AWS accounts

### Testing Guidelines

Tested with commercial accounts (sandbox). We currently don't have a gov AWS sandbox account so couldn't verify that


(This is a re-creation of https://github.com/DataDog/cloudformation-template/pull/100 after it was reverted from https://github.com/DataDog/cloudformation-template/pull/111. This should be fine now after adding versioning to our cloudformation templates from https://github.com/DataDog/cloudformation-template/pull/115. This means that this new `AWSPartition` Property will only affect new cloudformation stacks)
